packages/viewer-sandbox/src/main.ts

```diff
@@ -1,42 +1,32 @@
-import { Pane } from 'tweakpane'
-import { Viewer } from '@speckle/viewer'
+import { Viewer, IViewer } from '@speckle/viewer'
 import './style.css'
+import Sandbox from './Sandbox'
 
-const container = document.querySelector<HTMLDivElement>('#renderer')
+const container = document.querySelector<HTMLDivElement>('#renderer') as HTMLElement
```

Comment:
Don't forcefully recast types with `as` unless there's no other way, cause then 1) you leave the underlying typing problem unresolved 2) introduce the possibility of bugs where the TS compiler will think it's 1 type, but it's actually another.

In this case you can just change the generic type

Reply:
They are all HTMLElement

Reply:
Take a look at this example:

```ts
function fooBar(): any {
   // ...
   return someNumber;
}

// (in some other file)

const result = fooBar() as number;
```

For whatever reason the fooBar() function has an invalid return type and so you use `as` to cast it to number - because the value definitely is a number. Then later on someone rewrites the fooBar() function to this:

```ts
function fooBar(): bool {
   // ...
   return true;
}

// (in some other file)

const result = fooBar() as number;
```

The fooBar() function now has a completely different signature and yet TypeScript still thinks that `result` holds a number, because with `as` TypeScript was told to ignore whatever it think it is and to just use that forced type.

So forcefully casting a type to something other than what TypeScript thinks it is opens you up for more bugs in the future.
In this case it's unlikely that querySelector() would ever return something other than an HTMLElement, but being vary of using `as` is a good approach still and in this case you can easily specify the expected type in the generic argument of querySelector() without having to use `as`.

Reply:
I agree with the above. I never said that you can use `as` wheneve/wherever you want. However in this particular case, the use of `as` was very much harmless and also correct. `querySelector` does take a type argument here. My bad for not seeing that. However, in general, I don't rely much on typescript generics since they are just compile-time